opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java

```diff
@@ -237,17 +270,16 @@ public void testInsertTaskWithStartAndDue()
 
         ), resultsIn(mClient,
                 new Assert<>(task, new TimeData(start, due)),
-                new AssertRelated<>(new InstanceTable(mAuthority), Instances.TASK_ID, task, new AllOf(
-                        new EqArg(Instances.INSTANCE_START, start.getTimestamp()),
-                        new EqArg(Instances.INSTANCE_ORIGINAL_TIME, 0),
-                        new EqArg(Instances.INSTANCE_DUE, due.getTimestamp()),
-                        new EqArg(Instances.INSTANCE_START_SORTING, start.shiftTimeZone(TimeZone.getDefault()).getInstance()),
-                        new EqArg(Instances.INSTANCE_DUE_SORTING, due.shiftTimeZone(TimeZone.getDefault()).getInstance()),
-                        new EqArg(Instances.INSTANCE_DURATION, due.getTimestamp() - start.getTimestamp()),
-                        new EqArg(Tasks.TZ, start.isAllDay() ? "UTC" : start.getTimeZone().getID()),
-                        new EqArg(Instances.INSTANCE_STATUS, Instances.INSTANCE_STATUS_NEXT)
-                ))
-        ));
+                new AssertRelated<>(
+                        new InstanceTable(mAuthority), Instances.TASK_ID, task,
+                        new Composite<Instances>(
+                                new InstanceData(
+                                        start.shiftTimeZone(TimeZone.getDefault()),
```

Comment:
Why is it needed here to shift the time zone to the default? Aren't we supposed to keep the time zone as it is, which is `UTC` in this case?
The test passes without the shift.
If the shift is needed for the sorting columns, shouldn't that be applied inside `InstanceData`?
(I guess this applies to other test cases as well.)

Reply:
This is indeed about the sorting columns. IMO `InstanceData` should not convert the time zone. `InstanceData` is just not in the position to decide which time zone to use, there may be cases when you want to test explicitly for a time zone other than the default.

The test passes without shift only if your default timezone is `UTC`. If you change the timezone of your emulator the test should actually fail if remove the shift.

Reply:
In addition, sometimes the arguments to `InstanceData` need to be calculated, which has to be done in the local time zone (otherwise the result might be wrong), so we can't defer the shift to `InstanceData`.